Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,240 @@
+        return annotationViewsById[id]
+    }
+
+    /// Find `ViewAnnotationOptions` of view annotation by feature id if it was specified as part of `ViewAnnotationOptions.associatedFeatureId`.
+    ///
+    /// - Returns: `ViewAnnotationOptions` if view was found and `nil` otherwise.
+    public func options(byFeatureId identifier: String) -> ViewAnnotationOptions? {
+        return viewAnnotation(byFeatureId: identifier).flatMap { options(byAnnotationView: $0) }
+    }
+
+    /// Get current `ViewAnnotationOptions` for given `AnnotationView`.
+    ///
+    /// - Returns: `ViewAnnotationOptions` if view was found and `nil` otherwise.
+    public func options(byAnnotationView view: AnnotationView) -> ViewAnnotationOptions? {
+        return annotationIdByViews[view].flatMap { try? mapboxMap.options(forViewAnnotationWithId: $0) }
+    }
+
+    // MARK: - Internal functions
+
+    internal func placeAnnotations(positions: [ViewAnnotationPositionDescriptor]) {
+        // Iterate through and update all view annotations
+        // First update the position of the views based on the placement info from GL-Native
+        // Then hide the views which are off screen
+        var visibleAnnotationIds: Set<String> = []
+
+        for position in positions {
+            validateAnnotation(byAnnotationId: position.identifier)
+            guard let annotationView = annotationViewsById[position.identifier] else {
+                continue
+            }
+            annotationView.translatesAutoresizingMaskIntoConstraints = true
+            annotationView.frame = CGRect(
+                origin: position.leftTopCoordinate.point,
+                size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
+            )
+            annotationView.setInternalVisibility(isHidden: false)
+            visibleAnnotationIds.insert(position.identifier)
+        }
+
+        let annotationsToHide = Set<String>(annotationViewsById.keys).subtracting(visibleAnnotationIds)
+        for id in annotationsToHide {
+            validateAnnotation(byAnnotationId: id)
+            if let annotationView = annotationViewsById[id] {
+                annotationView.setInternalVisibility(isHidden: true)
+            }
+        }
+    }
+
+    internal func validateAnnotation(byAnnotationId id: String) {
+        guard let annotationView = annotationViewsById[id] else { return }
+        // If the user explicitly removed the ViewAnnotation or it's wrapped view
+        // we need to remove it from our layout calculation
+        if annotationView.subviews.isEmpty || annotationView.superview == nil {
+            try? remove(annotationView)
+        }
+        // View is still considered for layout calculation, users should not modify the visibility of the wrapped view
+        if let wrappedView = annotationView.subviews.first, wrappedView.isHidden {
+            Log.warning(forMessage: "Visibility changed for wrapped view", category: "Annotations")
```

Comment:
This warning message might need a bit more details. 

Also I was thinking to add a flag to disable this warning, which would allow the users to hide the wrapped views while they're still considered in the layout calculation. (Although the same could be achieved by setting the opacity to zero, so it might not be relevant.)